pkg/monitoring/metrics/virt-controller/vmstats_collector.go

```diff
@@ -410,3 +424,69 @@ func containsCondition(target k6tv1.VirtualMachineConditionType, elems []k6tv1.V
 	}
 	return false
 }
+
+func CollectDiskAllocatedSize(vms []*k6tv1.VirtualMachine) []operatormetrics.CollectorResult {
```

Comment:
this function should be refactored imo, it's too big
maybe try to move the body of the loops into separate function